File: app/src/main/java/app/endershrooms/thatcatapp/model/Category.java
package app.endershrooms.thatcatapp.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import com.squareup.moshi.Json;

@Entity
public class Category {

  @Json(name = "id")
  @PrimaryKey
  private Integer id;
  @Json(name = "name")
  private String name;

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

}

File: app/src/main/java/app/endershrooms/thatcatapp/model/Category.kt
package app.endershrooms.thatcatapp.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.squareup.moshi.Json

@Entity
data class Category(
    @Json(name = "id") @PrimaryKey var id: Int?,
    @Json(name = "name") var name: String?
)